packages/shell-api/src/database.ts

```diff
@@ -0,0 +1,155 @@
+import Mongo from './mongo';
+import Collection from './collection';
+import {
+  shellApiClassDefault,
+  returnsPromise,
+  returnType,
+  hasAsyncChild,
+  ShellApiClass,
+  serverVersions,
+} from './decorators';
+import { ServerVersions } from './enums';
+
+import {
+  Cursor as ServiceProviderCursor,
+  Document
+} from '@mongosh/service-provider-core';
+import { AggregationCursor } from './index';
+
+@shellApiClassDefault
+@hasAsyncChild
+export default class Database extends ShellApiClass {
+  mongo: Mongo;
+  name: string;
+
+  constructor(mongo, name) {
+    super();
+    this.mongo = mongo;
+    this.name = name;
+    const proxy = new Proxy(this, {
```

Comment:
Don't have much time to go through the rest of the code, this one i noticed previously, and it changed quite a bit in master, can you double check that we did not left bits behind?

Reply:
Yes, that's part of the rebasing in the TODO section of this PR's description

Reply:
This is the rebased PR: https://github.com/mongodb-js/mongosh/pull/195

Was complicated enough to make a new PR for the rebase, so closing this one.